src/utility/authentication-functions.js

```diff
@@ -27,7 +28,7 @@ const submitSignInDetails = async (
     });
     if (response.status === 200) {
       localStorage.setItem("token", response.data.encodedToken);
-      navigate("/");
+      navigate(location?.state?.from?.pathname, { replace: true });
```

Comment:
Well, I am not sure why do you need optional chaining on location anyway it going to return a value. you can replace this with  **location.state?.from.pathname** ..